### ISSUE TITLE
Game Crashes When Using "#goto" command in Baritone 1.20.1 Fabric Beta V1

### ISSUE THREAD

**Issue by Type-32** [cabaletta/baritone#4142]
## Some information
Operating system: Windows 11
Java version: JDK 17
Minecraft version: 1.20.1
Baritone version: baritone-api-fabric-1.20.1-beta-v1.jar
Other mods (if used): Sodium, iris, indium, etc.

## Exception, error or logs
[crash-2023-10-03_15.29.50-client.txt](https://github.com/cabaletta/baritone/files/12789759/crash-2023-10-03_15.29.50-client.txt)
[latest.log](https://github.com/cabaletta/baritone/files/12789760/latest.log)

## How to reproduce
1. Go on to a multiplayer server or a singleplayer world
2. Run the "#goto" command with any parameters (I specifically did "#goto minecraft:barrel", but inputting other parameters also crashes the game)
3. Game Crashes

## Modified settings
![image](https://github.com/cabaletta/baritone/assets/87076491/9ec634ba-98e1-4acf-b888-706f2d1ec0f6)

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [x] I have not used any OwO's or UwU's in this issue.

**Comment by ZacSharp** [comment]
Probably a duplicate of #4112 (also suspected in #4071)

Can you maybe reproduce this using an unoptimized build and post the log?

**Comment by Type-32** [comment]
_Stable Version: `baritone-api-fabric-1.10.1.jar` (From CurseForge)
Beta Version: `baritone-api-fabric-1.20.1-elytra-beta-v1.jar` (From Baritone Discord)_

I'm not sure where to get the unoptimized build for the version of baritone I'm using... the version I used is in the discord #announcement channel and is said to be a beta test.

However I used the stable posted version of baritone for 1.20.1 and that had no issues. It happens when I use the #goto command (in beta baritone), regardless of the block I'm trying to find, but in the stable release for 1.20.1 it doesn't crash and functions normally.

I checked the crash logs myself and is able to duplicate it on a singleplayer world: This seems to happen due to an `IndexOutOfBoundsException` being thrown, probably related to the function "scanChunkRadius()"

This bug seems to be exclusive to the beta baritone 1.20.1. I'm going to revert to using the stable version instead...

**Comment by Nekiplay** [comment]
[latest.log](https://github.com/cabaletta/baritone/files/12806567/latest.log)
log on lastest 1.20.1 commit, unoptimized

**Comment by ZacSharp** [comment]
I'm not sure from what commit the beta on the discord is built so you'd have to ask there.

The reason I asked for an unobfuscated log is that it would not only contain some random unobfuscated name 5 calls up the stack but also the name of the erroring function itself and even a precise line number where the exception happened.
Though that's not as important anymore since I'm now confident that this is the same bug as #4112 and the author of that issue did reproduce it using an unobfuscated build and it seems like they even found the reason for the exception.

Baritone does not have official CurseForge releases.

**Comment by Type-32** [comment]
![image](https://github.com/cabaletta/baritone/assets/87076491/2095eef7-19bd-4303-ab8a-de2c20dd69c1)
The image above is the version of the "beta version of baritone" I keep referring to (I am honestly not sure if I am allowed to upload the actual mod file here), which is already in discord.

Regarding the baritone version I got from curseforge, it is from [this page here](https://www.curseforge.com/minecraft/mc-mods/baritone-1-18-2) which I am fully aware of that it is not an official page of baritone but my diction is just incorrect.

I rechecked (checked it two times before but missed some stuff) again the issue you had mentioned and the logs that that player had gave looks similar to of what I had.

Thanks for the help. :)
